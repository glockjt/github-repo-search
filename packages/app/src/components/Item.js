import PropTypes from "prop-types";
/** @jsx jsx */
import { css, jsx } from "@emotion/core";
import styled from "@emotion/styled";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
import colors from "../colors.json";

const StyledItem = styled.a`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  width: 100%;
  border-bottom: 2px solid hsl(210, 23%, 95%);
  box-sizing: border-box;
  padding: 16px 8px;
  text-decoration: none;
  color: black;
  background-color: #fff;

  :hover {
    cursor: pointer;
    background-color: #d9e2ec;
  }
`;

const Name = styled.div`
  color: #102a43;
  font-weight: 600;
  flex: 2;
`;

const Description = styled.div`
  color: #829ab1;
  flex: 1;
  margin-top: 8px;
`;

const Avatar = styled.img`
  width: 25px;
  border-radius: 50%;
  margin-right: 4px;
`;

const Owner = styled.div`
  color: #102a43;
  display: flex;
  align-items: center;
  color: dark-gray;
  flex: 2;
`;

const Language = styled.div`
  color: #829ab1;
  display: flex;
  flex: 1;
  align-items: center;
`;

const LanguageAccent = styled.div`
  height: 16px;
  margin-right: 4px;
  border: 1px solid
    ${props => (colors[props.language] ? colors[props.language].color : "")};
`;

const Stars = styled.div`
  color: #829ab1;
  display: flex;
  font-weight: 300;
  flex: 0;
  min-width: 80px;
  align-items: center;
  padding-left: 8px;
`;

const Item = props => (
  <StyledItem href={props.url} target="__blank">
    <div
      css={css`
        display: flex;
        flex: 1;
        align-items: center;
        @media (max-width: 425px) {
          flex-direction: column;
          align-items: flex-start;
        }
      `}
    >
      <Name>{props.name}</Name>
      <div
        css={css`
          display: flex;
          flex: 3;
          @media (max-width: 425px) {
            display: flex;
            flex: 1;
            justify-content: space-between;
            width: 100%;
            margin-top: 16px;
          }
        `}
      >
        {props.owner && (
          <Owner>
            <Avatar src={props.owner.avatar_url} alt="" />
            <div>{props.owner.login}</div>
          </Owner>
        )}
        <Language>
          {props.language && <LanguageAccent language={props.language} />}
          {props.language}
        </Language>
        <Stars>
          <FontAwesomeIcon
            css={css`
              color: #0e7c86;
              font-size: 12px;
              margin-right: 8px;
            `}
            icon={faStar}
          />
          {props.stars}
        </Stars>
      </div>
    </div>
    <Description>{props.description}</Description>
  </StyledItem>
);

Item.propTypes = {
  url: PropTypes.string,
  name: PropTypes.string,
  owner: PropTypes.shape({
    avatar_url: PropTypes.string,
    login: PropTypes.string
  }),
  language: PropTypes.string,
  stars: PropTypes.number,
  description: PropTypes.string
};

export default Item;
