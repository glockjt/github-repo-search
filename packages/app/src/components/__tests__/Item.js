import React from "react";
import renderer from "react-test-renderer";
import Item from "../Item";

describe("Item tests", () => {
  it("should render empty item", () => {
    const tree = renderer.create(<Item />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render a fully populated item", () => {
    const tree = renderer
      .create(
        <Item
          name="test"
          url="testurl.com"
          owner={{ avatar_url: "avatar.com", login: "login" }}
          language="JavaScript"
          stars={100}
          description="Testing"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render an item with no owner", () => {
    const tree = renderer
      .create(
        <Item
          name="test"
          url="testurl.com"
          language="JavaScript"
          stars={100}
          description="Testing"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
