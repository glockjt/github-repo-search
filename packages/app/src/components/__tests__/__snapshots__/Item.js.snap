// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item tests should render a fully populated item 1`] = `
<a
  className="css-arizad"
  href="testurl.com"
  target="__blank"
>
  <div
    className="css-zjwx56-Item"
  >
    <div
      className="css-1f1qa1o"
    >
      test
    </div>
    <div
      className="css-1npim7z-Item"
    >
      <div
        className="css-adtgxk"
      >
        <img
          alt=""
          className="css-e4eikt"
          src="avatar.com"
        />
        <div>
          login
        </div>
      </div>
      <div
        className="css-iazxl9"
      >
        <div
          className="css-10lwzoi"
        />
        JavaScript
      </div>
      <div
        className="css-1cyy2lc"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 css-1gkn1fq-Item"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        100
      </div>
    </div>
  </div>
  <div
    className="css-1y4ob8d"
  >
    Testing
  </div>
</a>
`;

exports[`Item tests should render an item with no owner 1`] = `
<a
  className="css-arizad"
  href="testurl.com"
  target="__blank"
>
  <div
    className="css-zjwx56-Item"
  >
    <div
      className="css-1f1qa1o"
    >
      test
    </div>
    <div
      className="css-1npim7z-Item"
    >
      <div
        className="css-iazxl9"
      >
        <div
          className="css-10lwzoi"
        />
        JavaScript
      </div>
      <div
        className="css-1cyy2lc"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 css-1gkn1fq-Item"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        100
      </div>
    </div>
  </div>
  <div
    className="css-1y4ob8d"
  >
    Testing
  </div>
</a>
`;

exports[`Item tests should render empty item 1`] = `
<a
  className="css-arizad"
  target="__blank"
>
  <div
    className="css-zjwx56-Item"
  >
    <div
      className="css-1f1qa1o"
    />
    <div
      className="css-1npim7z-Item"
    >
      <div
        className="css-iazxl9"
      />
      <div
        className="css-1cyy2lc"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 css-1gkn1fq-Item"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="css-1y4ob8d"
  />
</a>
`;
